src/player.cpp

```diff
@@ -4717,6 +4717,22 @@ dealt_damage_instance player::deal_damage(Creature* source, body_part bp, const
         }
     }
 
+        //Acid blood effects.
+        bool u_see = g->u.sees(*this);
+        if (has_trait("ACIDBLOOD") && !one_in(3) && (dam >= 1)) {
+            if (is_player()) {
+                add_msg(m_good, _("Your acidic blood splashes %s in mid-attack!"),
```

Comment:
Probably should have a range check.  NPC shot me (by mistake) from 3-4 tiles away and got acid'd.